src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -153,32 +154,30 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
     }
 
     private async _getPayment(methodId: string): Promise<PaymentMethodData> {
-        if (!methodId) {
+        if (!methodId || !this._paymentMethod) {
             throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
-        let state = this._store.getState();
-        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
-
-        const { nonce } = this._paymentMethod.initializationData;
-        if (nonce) {
-            state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
-            this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
-        }
-
-        const { card_information: cardInformation } = this._paymentMethod.initializationData;
-
         return {
             methodId,
             paymentData: {
                 method: methodId,
-                cardInformation,
-                nonce: this._getNonce(methodId, this._paymentMethod),
+                cardInformation: this._paymentMethod.initializationData.card_information,
+                nonce: await this._getNonce(methodId),
             },
         };
     }
 
-    private _getNonce(methodId: string, { initializationData: { nonce }}: PaymentMethod) {
+    private async _getNonce(methodId: string) {
+        if (!this._paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+        const nonce = this._paymentMethod.initializationData.nonce;
+
+        if (this._paymentMethod.initializationData.nonce) {
+            this._paymentMethod = { ...this._paymentMethod,  initializationData: { nonce: ''} };
+        }
```

Comment:
I don't think this method should be cleaning the used nonce. That should probably live somewhere else. The caller maybe?